### PR TITLE
Update solution branch to reflect the changes to master

### LINKED ISSUE #17: Dogs tests not hinted at or explained at all in the README.md
So this lab is really quite confusing. I created a Dog class as a model that inherited from `model.rb`... assuming that's what they were looking for... anyway I got the tests to pass, and when I looked at the solution code after submitting I saw that they created an `@dog` instance var to pass to the erb file from `ApplicationController`... but this is a poorly designed lesson. I have a little background in rails and MVC pattern so I figured they were looking for a model (though there is no test nor hint that they want to create an instance to pass to the view here)... some better wording is needed to explain this random request for a simple Dog class... at least a hint that it should be a model or something. I can imagine a lot of students being totally lost on this, or creating code that passes but doesn't necessarily serve the purpose of introducing the idea of passing data from a model to controller to view...

### PR DESCRIPTION
**Pull request by mendelB**
The master branch has been drastically changed to be much less of a code along to have more things pre-filled/not tested. This will update the solution as well and fix #17 and reflect #18 as well
@PeterBell @AnnJohn

### CODE REVIEW

**Comment by PeterBell** [comment]
Really big rewrite so I'd love to see input from @jmburges just to make sure we're taking this the right way.

**Comment by AnnJohn** [comment]
for web-0416; @pletcher can you also give this a review? Closes #17 and closes #18

File: README.md [hunk 1]
```diff
@@ -2,15 +2,379 @@
...
+    ├── features
+    ├── models
+    └── spec_helper.rb
+```
+
+### `Gemfile`
+
+This holds a list of all the gems needed to run the application. The bundler gem provides us access to a terminal command: `bundle install`. Bundler will look in the Gemfile and install any gems, as well as any gem dependencies for this application.
+
+Go ahead and enter this command in terminal. It will create a `Gemfile.lock` file for you, which is just a documentation of what versions of the gem you have installed and should use. The lock word is actually because it makes sure that only one thing is running bundle install at a time.
```
Comment (pletcher) [inline README.md:51]: This isn't quite right; `Gemfile.lock` _locks down_ the versions of gem dependencies more strictly than a Gemfile alone.

From the docs:

> This is important: the Gemfile.lock makes your application a single package of both your own code and the third-party code it ran the last time you know for sure that everything worked.

http://bundler.io/v1.3/rationale.html

File: README.md [hunk 2]
```diff
@@ -2,15 +2,379 @@
...
+
+### `app` directory
+
+This folder holds our MVC directories - `models`, `views`, and `controllers`. We spend most of our time coding in this directory.
+
+#### `models` directory
+
+This directory holds the logic behind our application. Typically, these files represent either a component of your application, such as a User, Post, or Comment, or a unit of work. Each file in models typically contains a different class. For example, `dog.rb` would contain a class called `Dog`. As you might have guessed, models represent the "M" components of the MVC paradigm.
+
+Models represent the data and object logic of our application.
```
Comment (pletcher) [inline README.md:61]: This seems to repeat the first sentence in the paragraph above.

File: README.md [hunk 3]
```diff
@@ -2,15 +2,379 @@
...
+
+#### `controllers` directory
+
+The controllers, such as `application_controller.rb`, are where the application configurations, routes, and controller actions are implemented. There is typically a class, which in this case we will call `ApplicationController`, that represents an instance of your application when the server is up and running. The `application_controller.rb` file represents the "C" components of the MVC paradigm.
+
+(In some simple applications -including several labs and code-alongs in this track - the Application Controller will simply be called `app.rb` and will live in the root directory of the project)
+
+Sometimes our other controllers will use `ApplicationController` as an inheritance point so that they inherit all the defaults and behaviors defined in our main `ApplicationController`. Other times our other controllers will simply inherit from `Sinatra::Base`.
```
Comment (pletcher) [inline README.md:71]: Could the first sentence state more simply, "Sometimes our other controllers will inherit from `ApplicationController`."?

File: README.md [hunk 4]
```diff
@@ -2,15 +2,379 @@
...
+
+Controllers represent the application logic, generally, the interface and flow of our application.
```
Comment (pletcher) [inline README.md:73]: We said above that models represent the application logic -- could we make the distinction between models and controllers clearer?

File: README.md [hunk 5]
```diff
@@ -2,15 +2,379 @@
...
+
+Once all our code is loaded, `config.ru` then specifies which controllers to load as part of our application using `run` or `use`.
+
+In this case, our `config.ru` file has this line `run ApplicationController`, creating an instance of our ApplicationController class which can respond to requests from a client.
+
+### `config` directory
+
+This directory holds an `environment.rb` file. We'll be using this file to connect up all the files in our application to the appropriate gems and to each other.
+
+This `environment.rb` file loads Bundler and thus all the gems in our Gemfile, as well as the `app` directory`.
```
Comment (pletcher) [inline README.md:106]: You've got an extra backtick at the end of this line :)

File: README.md [hunk 6]
```diff
@@ -2,15 +2,379 @@
...
+The `public` directory holds our front-end assets. In the example above, it holds a `css` directory with a stylesheet. Javascript directories and any other front-end assets (like image files) should also be stored in `public`.
+
+### `spec` directory
+
+The `spec` directory contains any tests for our applications. These tests set up any expectations for the rest of the project. These are often broken down into unit tests for models, controller tests for routes, and feature tests, which check the actual behavior for users.
+
+Don't forget to run your tests!
+
+<p data-visibility='hidden'>View <a href='https://learn.co/lessons/sinatra-mvc-file-structure' title='Sinatra MVC File Structure'>Sinatra MVC File Structure</a> on Learn.co and start learning to code for free.</p>
+# Sinatra MVC File Structure
```
Comment (pletcher) [inline README.md]: Looks like we should delete everything below this line?

**Comment by pletcher** [comment]
Something funny happened in the README -- the text looks like it repeats 3 or 4 times? -- and I left a few comments to clear some things up.

**Comment by mendelB** [comment]
Just to clarify: _I did not write this Readme_ :) I simply copied the Readme over from the master branch - for some reason the solutions Readme was almost the same, with some minor but crucial differences - the only change I made was a clarification at the end of the models paragraph to address #17 .
(this pull request was just to have the solution branch reflect the master one as well as #18 where the change was made to that models paragraph and a small fix to the spec.)
@pletcher

**Comment by pletcher** [comment]
Haha, no worries -- I figured something like that had happened.

If this is just to sync things up, I think we can go ahead and merge it. But we should definitely circle back and fix the description of Gemfile.lock, etc.

**Pull request merged**